src/overlays/actors/ovl_Eff_Dust/z_eff_dust.c

```diff
@@ -22,28 +29,392 @@ const ActorInit Eff_Dust_InitVars = {
+    }
+
+    for (i = 0; i < 0x40; i++) {
+        if ((*distanceTraveled) < 1.0f) {
+            *distanceTraveled += 0.1f;
+        }
+        distanceTraveled++;
+    }
+
+    this->actor.world.pos = parent->world.pos;
+
+    for (j = 0; j < 3; j++) {
+        i = this->index & 0x3F;
+        if (this->distanceTraveled[i] >= 1.0f) {
+
+            theta = Rand_CenteredFloat(65536.0f);
+            switch (this->actor.params) {
+                case Effect_2:
+                    this->initialPositions[i].x = (Rand_ZeroOne() * 4500.0f) + 700.0f;
+                    if (this->initialPositions[i].x > 3000.0f) {
+                        this->initialPositions[i].y = (3000.0f * Rand_ZeroOne()) * Math_SinS(theta);
+                        this->initialPositions[i].z = (3000.0f * Rand_ZeroOne()) * Math_CosS(theta);
+                    } else {
+                        this->initialPositions[i].y = 3000.0f * Math_SinS(theta);
+                        this->initialPositions[i].z = 3000.0f * Math_CosS(theta);
+                    }
+                    break;
+
+                case Effect_3:
+                    this->initialPositions[i].x = (Rand_ZeroOne() * 2500.0f) + 700.0f;
+                    if (this->initialPositions[i].x > 2000.0f) {
+                        this->initialPositions[i].y = (2000.0f * Rand_ZeroOne()) * Math_SinS(theta);
+                        this->initialPositions[i].z = (2000.0f * Rand_ZeroOne()) * Math_CosS(theta);
+                    } else {
+                        this->initialPositions[i].y = 2000.0f * Math_SinS(theta);
+                        this->initialPositions[i].z = 2000.0f * Math_CosS(theta);
+                    }
+                    break;
+
+                case Effect_4:
+                    this->initialPositions[i].x = (Rand_ZeroOne() * 8500.0f) + 1700.0f;
+                    if (this->initialPositions[i].x > 5000.0f) {
+                        this->initialPositions[i].y = (4000.0f * Rand_ZeroOne()) * Math_SinS(theta);
+                        this->initialPositions[i].z = (4000.0f * Rand_ZeroOne()) * Math_CosS(theta);
+                    } else {
+                        this->initialPositions[i].y = 4000.0f * Math_SinS(theta);
+                        this->initialPositions[i].z = 4000.0f * Math_CosS(theta);
+                    }
+
+                    break;
+                default:
+                    break;
+            }
+
+            this->distanceTraveled[i] = 0.0f;
+            this->index += 1;
+        }
+    }
+}
+
+void EffDust_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EffDust* this = THIS;
+    this->updateFunc(this, globalCtx);
+}
+
+void EffDust_DrawFunc_8099E4F4(EffDust* this, GlobalContext* globalCtx) {
+    EffDust* this2;
+
+    GlobalContext* glb_ctx;
+    GraphicsContext* gfx_ctx;
+
+    Vec3f* initialPositions;
+    f32* distanceTraveled;
+
+    s32 i;
+
+    f32 aux;
+
+    glb_ctx = globalCtx;
+    gfx_ctx = globalCtx->state.gfxCtx;
+    this2 = this;
```

Comment:
You might be able to avoid this if you pass in `Actor* thisx` since this is a draw function its possible thats whats needed.